### PR TITLE
Replace deprecated MGConstrainedDoFs::initialize in examples

### LINKED ISSUE #2923: Handle deprecated MGConstrainedDoFs::initialize() in examples
Step 16, 50, and 56 make use of MGConstrainedDoFs::initialize(dof_handler, function_map, component_mask) which is marked as deprecated.

The accompanying note in the documentation suggests the replacement should be initialize(dof_handler) followed by set_zero_boundary_dofs. The second of these two functions does not exist, and make_zero_boundary_dofs has a significantly differing signature and apparent intended use.

### PR DESCRIPTION
**Pull request by class4kayaker**
Fixes #2923.

Begin working on fixing calls to deprecated MGConstrainedDoFs::initialize(dof, fn_map) in examples. The main necessary replacement has been made, but documentation may need to be updated and some of the logic needs to be checked to see if some lines can be removed due to no longer being required.

### CODE REVIEW

**Comment by masterleinad** [comment]
Would you also replace `Use initialize() followed by set_zero_boundary_dofs() instead` by `Use initialize() followed by make_zero_boundary_constraints()` in the documentation for `MGConstrainedDoFs` and thus closing #2932 entirely?

File: examples/step-56/step-56.cc [hunk 1]
```diff
@@ -554,12 +554,15 @@ namespace Step56
         // sparsity patterns and matrices for each level. The resize()
         // function of MGLevelObject<T> will destroy all existing contained
         // objects.
+        std::set<types::boundary_id> zero_boundary_ids;
         typename FunctionMap<dim>::type boundary_condition_function_map;
```
Comment (masterleinad) [inline examples/step-56/step-56.cc]: As far as I see, this object is not used any more and can be removed.

Comment (class4kayaker) [inline examples/step-56/step-56.cc]: Yes, this one is not used elsewhere, I assumed that the pattern I had noticed with the other two was repeated, and it was still used.

**Comment by masterleinad** [comment]
Looks good! Is there anything you wanted to address here and didn't so far?

**Comment by class4kayaker** [comment]
I do not have anything else I was looking to modify. The only question I had was whether a modification of the example documentation was necessary, but I did not see any obvious places where the modification has changed the necessary description.

**Comment by masterleinad** [comment]
Thanks for looking into this!

**Pull request merged**